components/eam/src/chemistry/mozart/chemistry.F90

```diff
@@ -1516,10 +1540,84 @@ subroutine chem_timestep_tend( state, ptend, cam_in, cam_out, dt, pbuf,  fh2o, f
+             end do
+             do k= gaschmbudget_2D_L2_s, gaschmbudget_2D_L2_e ! 90-300 hPa
+               ftem_layers(:ncol,2) = ftem_layers(:ncol,2) + ftem(:ncol,k)
+               gas_ac_layers(:ncol,2) = gas_ac_layers(:ncol,2) + gas_ac(:ncol,k)
+             end do
+             do k= gaschmbudget_2D_L3_s, gaschmbudget_2D_L3_e ! 300-850 hPa
+               ftem_layers(:ncol,3) = ftem_layers(:ncol,3) + ftem(:ncol,k)
+               gas_ac_layers(:ncol,3) = gas_ac_layers(:ncol,3) + gas_ac(:ncol,k)
+             end do
+             do k= gaschmbudget_2D_L4_s, gaschmbudget_2D_L4_e ! 850 hPa - surface
+               ftem_layers(:ncol,4) = ftem_layers(:ncol,4) + ftem(:ncol,k)
+               gas_ac_layers(:ncol,4) = gas_ac_layers(:ncol,4) + gas_ac(:ncol,k)
+             end do
+             call outfld(trim(solsym(n))//'_2DMSB_L1', ftem_layers(:ncol,1), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DMSB_L2', ftem_layers(:ncol,2), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DMSB_L3', ftem_layers(:ncol,3), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DMSB_L4', ftem_layers(:ncol,4), pcols, lchnk)
+
+             if( nstep == 0 ) then
+                Diff_layers(:ncol,:) = 0.0_r8
+             else
+                Diff_layers(:ncol,:) = 0.0_r8
+                Diff_layers(:ncol,1) = (ftem_layers(:ncol,1) - gas_ac_layers(:ncol,1))/dt
+                Diff_layers(:ncol,2) = (ftem_layers(:ncol,2) - gas_ac_layers(:ncol,2))/dt
+                Diff_layers(:ncol,3) = (ftem_layers(:ncol,3) - gas_ac_layers(:ncol,3))/dt
+                Diff_layers(:ncol,4) = (ftem_layers(:ncol,4) - gas_ac_layers(:ncol,4))/dt
+             end if
+             call outfld(trim(solsym(n))//'_2DTDO_L1', Diff_layers(:ncol,1), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DTDO_L2', Diff_layers(:ncol,2), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DTDO_L3', Diff_layers(:ncol,3), pcols, lchnk)
+             call outfld(trim(solsym(n))//'_2DTDO_L4', Diff_layers(:ncol,4), pcols, lchnk)
+
+             if (trim(solsym(n))=='O3' .or. trim(solsym(n))=='O3LNZ' .or. &
+                 trim(solsym(n))=='N2OLNZ' .or. trim(solsym(n))=='CH4LNZ') then
+                ftem_layers = 0.0_r8
+                gas_ac_layers = 0.0_r8
+                do k=1,pver
+                   ftem_layers(:ncol,1) = ftem_layers(:ncol,1) + ftem(:ncol,k) * tropFlagInt(:ncol,k)
+                   gas_ac_layers(:ncol,1) = gas_ac_layers(:ncol,1) + gas_ac(:ncol,k) * tropFlagInt(:ncol,k)
+                   ftem_layers(:ncol,2) = ftem_layers(:ncol,2) + ftem(:ncol,k) * (1.0 - tropFlagInt(:ncol,k))
+                   gas_ac_layers(:ncol,2) = gas_ac_layers(:ncol,2) + gas_ac(:ncol,k) * (1.0 - tropFlagInt(:ncol,k))
+                end do
+                call outfld(trim(solsym(n))//'_2DMSB_trop', ftem_layers(:ncol,1), pcols, lchnk )
+                call outfld(trim(solsym(n))//'_2DMSB_stra', ftem_layers(:ncol,2), pcols, lchnk )
```

Comment:
Is the stratospheric output ('*_stra') for testing purpose?

Reply:
I removed "_stra" in the code.